### PR TITLE
Make device parameter for area selector optional

### PR DESCRIPTION
**Pull request by raman325**
## Proposed change
The `device` key in the area selector config is optional

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by c0ffeeca7 — APPROVED** [review]
:star_struck: Thank you, @raman325 :+1:

File: source/_docs/blueprint/selectors.markdown [hunk 1]
```diff
@@ -18,27 +18,32 @@ The following selectors are currently available:
 - [Action selector](#action-selector)
 - [Add-on selector](#add-on-selector)
 - [Area selector](#area-selector)
+  - [Example area selectors](#example-area-selectors)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown:21]: @raman325 @c0ffeeca7 

This not a traditional TOC, seems this unrelated change sneaked in. Could you take a look?

File: source/_docs/blueprint/selectors.markdown [hunk 2]
```diff
@@ -18,27 +18,32 @@ The following selectors are currently available:
...
 - [Assist pipeline selector](#assist-pipeline-selector)
 - [Boolean selector](#boolean-selector)
 - [Color temperature selector](#color-temperature-selector)
 - [Config entry selector](#config-entry-selector)
 - [Constant selector](#constant-selector)
 - [Date selector](#date-selector)
-- [Date & time selector](#date--time-selector)
+- [Date \& time selector](#date--time-selector)
 - [Device selector](#device-selector)
+  - [Example device selector](#example-device-selector)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown:31]: Same

File: source/_docs/blueprint/selectors.markdown [hunk 3]
```diff
@@ -18,27 +18,32 @@ The following selectors are currently available:
...
 - [Duration selector](#duration-selector)
 - [Entity selector](#entity-selector)
+  - [Example entity selector](#example-entity-selector)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown:34]: Same

File: source/_docs/blueprint/selectors.markdown [hunk 4]
```diff
@@ -18,27 +18,32 @@ The following selectors are currently available:
...
 - [Icon selector](#icon-selector)
 - [Language selector](#language-selector)
 - [Location selector](#location-selector)
 - [Media selector](#media-selector)
 - [Number selector](#number-selector)
+  - [Example number selectors](#example-number-selectors)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown:40]: Same

File: source/_docs/blueprint/selectors.markdown [hunk 5]
```diff
@@ -18,27 +18,32 @@ The following selectors are currently available:
...
 - [Object selector](#object-selector)
 - [RGB color selector](#rgb-color-selector)
 - [Select selector](#select-selector)
 - [State selector](#state-selector)
 - [Target selector](#target-selector)
+  - [Example target selectors](#example-target-selectors)
```
Comment (frenck) [inline source/_docs/blueprint/selectors.markdown:46]: Same

File: source/_docs/blueprint/selectors.markdown (hunk 1 above)
Comment (c0ffeeca7) [reply source/_docs/blueprint/selectors.markdown:21]: Right. Let me remove it.

Comment (raman325) [reply source/_docs/blueprint/selectors.markdown:21]: Hmm I'm not sure how this happened. Perhaps I started from next and then switched to current? Not sure
